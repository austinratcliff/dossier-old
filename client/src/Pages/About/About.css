.About {
  position: absolute;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  background-color: white;
  color: #111;
  font-family: 'IBM Plex Mono', monospace;
  height: 100vh;
  width: 100vw;
  max-width: none !important;
}

.About .v-center-header {
  position: absolute;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
  width: 100vw;
}

.About .header {
  position: absolute;
  font-size: calc(10px + 5vmin);
  margin: 0;
  animation: slide-up 1.5s 1.5s 1 forwards;
}

.About .austin {
  border: 5px solid #111;
  margin-top: 15px;
  opacity: 0;
  visibility: hidden;
  animation: fade-in 1s 2s 1 forwards;
}

.About .text {
  text-align: left;
  margin-top: 15px;
  opacity: 0;
  visibility: hidden;
  animation: fade-in 1s 2s 1 forwards;
}

.About .description {
  margin: 0;
}

.About .bio {
  line-height: 1.5;
  margin: 15px 0 0 0;
}

.About .home {
  margin: 15px 0;
  cursor: pointer;
}

.About .highlight {
  background: linear-gradient(180deg, rgba(255, 255, 255, 0) 50%, #92A8D1 50%);
  padding: 0 5px;
}

@media (min-width: 576px) and (max-width: 991px) {
  .About .austin {
    width: 50vw;
  }

  .About .text {
    width: 50vw;
  }
}

@media (min-width: 992px) {
  .About {
    flex-direction: row;
    justify-content: center;
    overflow: hidden;
  }

  .About .austin {
    width: 35vw;
    margin: 0 15px;
  }

  .About .text {
    width: 40vw;
    margin: 0 15px;
  }

  .About .description {
    font-size: calc(10px + 2vmin);
  }

  .About .bio {
    font-size: calc(10px + 1vmin);
    text-align: justify;
  }

  .About .home {
    position: absolute;
    bottom: 0;
    margin-bottom: 25px;
    animation: bounce-down 1s infinite alternate;
  }

  .About .home:hover {
    animation-play-state: paused;
  }
}

@media (min-width: 1200px) {
  .About .austin {
    position: absolute;
    transform: translate(-15vw, -15vh);
    width: 25vw;
    animation: fade-in-down 1s 2s 1 forwards;
  }

  .About .text {
    position: absolute;
    transform: translate(15vw, 15vh);
    width: 30vw;
    animation: fade-in-up 1s 2s 1 forwards;
  }
}

@keyframes slide-up {
  from {transform: translateY(0);}
  to {transform: translateY(-100vh);}
}

@keyframes fade-in {
  from {opacity: 0; visibility: hidden;}
  to {opacity: 1; visibility: visible;}
}

@keyframes fade-in-down {
  from {transform: translate(-15vw, -15vh); opacity: 0; visibility: hidden;}
  to {transform: translate(-15vw, -10vh); opacity: 1; visibility: visible;}
}

@keyframes fade-in-up {
  from {transform: translate(15vw, 15vh); opacity: 0; visibility: hidden;}
  to {transform: translate(15vw, 10vh); opacity: 1; visibility: visible;}
}

@keyframes bounce-down {
  from {margin-bottom: 25px;}
  to {margin-bottom: 15px;}
}
