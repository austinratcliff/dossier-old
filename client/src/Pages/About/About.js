import React from 'react'
import Container from 'react-bootstrap/Container'
import Image from 'react-bootstrap/Image'
import austin from './assets/austin.png'
import './About.css'

const About = props => {
  return (
    <Container className="About">
      <div className="v-center-header">
        <h1 className="header">
          <i
            className="fas fa-user"
            title="about"
          /> About
        </h1>
      </div>
      <Image
        className="austin"
        src={austin}
        alt="austin"
        fluid
      />
      <div className="text">
        <h2 className="description">
          <span className="highlight">
            Habitually curious.
          </span>
        </h2>
        <h2 className="description">
          <span className="highlight">
            Strategically detailed.
          </span>
        </h2>
        <h2 className="description">
          <span className="highlight">
            Politely daring.
          </span>
        </h2>
        <p className="bio">
          Austin is an Oklahoma-born, Chicago-living software developer with an affinity for clean code, astrology, and oversized sweatshirts. Originally a fashion marketing and journalism grad, his dream of being the next Anna Wintour changed after he learned web development at Dev Bootcamp. Now, when he's not at work, he can be found at his favorite coffee shop in Wicker Park, either working on yet another iteration of one of his side projects or online shopping.
        </p>
      </div>
      <i
        className="fas fa-angle-up fa-2x home"
        title="home"
        onClick={props.handleClick}
      />
    </Container>
  )
}

export default About
