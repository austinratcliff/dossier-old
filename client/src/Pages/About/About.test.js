import React from 'react'
import ReactDOM from 'react-dom'
import ReactTestRenderer from 'react-test-renderer'
import About from './About'

describe('About component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<About />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('matches snapshot', () => {
    const component = ReactTestRenderer.create(<About />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
