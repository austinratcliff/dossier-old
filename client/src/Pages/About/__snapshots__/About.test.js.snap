// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component matches snapshot 1`] = `
<div
  className="About container"
>
  <div
    className="v-center-header"
  >
    <h1
      className="header"
    >
      <i
        className="fas fa-user"
        title="about"
      />
       About
    </h1>
  </div>
  <img
    alt="austin"
    className="austin img-fluid"
    src="austin.png"
  />
  <div
    className="text"
  >
    <h2
      className="description"
    >
      <span
        className="highlight"
      >
        Habitually curious.
      </span>
    </h2>
    <h2
      className="description"
    >
      <span
        className="highlight"
      >
        Strategically detailed.
      </span>
    </h2>
    <h2
      className="description"
    >
      <span
        className="highlight"
      >
        Politely daring.
      </span>
    </h2>
    <p
      className="bio"
    >
      Austin is an Oklahoma-born, Chicago-living software developer with an affinity for clean code, astrology, and oversized sweatshirts. Originally a fashion marketing and journalism grad, his dream of being the next Anna Wintour changed after he learned web development at Dev Bootcamp. Now, when he's not at work, he can be found at his favorite coffee shop in Wicker Park, either working on yet another iteration of one of his side projects or online shopping.
    </p>
  </div>
  <i
    className="fas fa-angle-up fa-2x home"
    title="home"
  />
</div>
`;
